### PR TITLE
Merge develop changes into main

### PR DESCRIPTION
**Pull request by breyerml**
New and improved features:
- add a new HIP backend (currently only tested on AMD GPUs)
- add custom OpenCL kernel binary caching
- now both DPC++ and hipSYCL can be used in a single executable
- slightly improved IO performance
- fix a performance bug in the OpenMP backend
- other small QoL changes: automatic `--backend` and `--target_platform`, automatic `PLSSVM_TARGET_PLATFORMS` deduction in CMake, ...
- bugfixes

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -140,23 +151,26 @@ else()
             $<BUILD_INTERFACE:${fast_float_SOURCE_DIR}/include>
             $<INSTALL_INTERFACE:include>
             )
+    message(STATUS "Installed fast_float version ${PLSSVM_fast_float_VERSION}.")
 endif()
 
 ## try finding fmt
-find_package(fmt QUIET)
+find_package(fmt 8.0.0 QUIET)
```
Comment (vancraar) [inline CMakeLists.txt:158]: Have we checked if this is the minimum version, or is it any that PLSSVM currently runs with?

File: .jenkins/Jenkinsfile-Tests-Multi-GPU [hunk 2]
```diff
@@ -138,36 +180,38 @@ pipeline {
                         }
                     }
                 }
-                stage('hipSYCL'){
+                // hipSYCL backend
+                stage('Build and Test - hipSYCL'){
                     stages{
-                        stage('build plssvm hipSYCL Release') {
+                        stage('Build PLSSVM hipSYCL - OpenCL') {
```
Comment (vancraar) [inline .jenkins/Jenkinsfile-Tests-Multi-GPU:186]:
```suggestion
                        stage('Build PLSSVM Release - hipSYCL') {
```

File: .jenkins/Jenkinsfile-Tests-Multi-GPU [hunk 3]
```diff
@@ -259,9 +304,9 @@ pipeline {
                     --header "authorization: Bearer ${gitlab_token}" \
                     --data "{
                         \\"state\\": \\"success\\",
-                        \\"context\\": \\"jenkins-ctest-multigpu\\",
-                        \\"description\\": \\"Jenkins CI Job: jenkins-ctest-multigpu\\",
-                        \\"target_url\\": \\"https://simsgs.informatik.uni-stuttgart.de/jenkins/view/PLSSVM/job/PLSSVM/job/Github-Multigpu/job/${BRANCH_NAME}/$BUILD_NUMBER\\"
+                        \\"context\\": \\"Jenkins Multi-GPU tests\\",
+                        \\"description\\": \\"Jenkins CI Job: Jenkins Multi-GPU testsu\\",
```
Comment (vancraar) [inline .jenkins/Jenkinsfile-Tests-Multi-GPU:308]:
```suggestion
                        \\"description\\": \\"Jenkins CI Job: Jenkins Multi-GPU tests\\",
```

File: include/plssvm/backends/HIP/predict_kernel.hip.hpp [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        temp += alpha_d[num_data_points - 1] * data_last_d[index];
+        w_d[index] = temp;
+    }
+}
+
+/**
+ * @brief Predicts the labels for data points using the polynomial kernel function.
+ * @details Currently only single GPU execution is supported.
+ * @tparam real_type the type of the data
+ * @param[in] out_d the calculated predictions
```
Comment (vancraar) [inline include/plssvm/backends/HIP/predict_kernel.hip.hpp:49]:
```suggestion
 * @param[out] out_d the calculated predictions
```

**Review by vancraar — COMMENTED** [review]
In addition, make it clearer what was generated automatically, or what is automatically overwritten and therefore should not be edited manually.

File: docs/plssvm-train.1 [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+.B -c, --cost arg
+set the parameter C (default: 1)
+
+.TP
+.B -e, --epsilon arg
+set the tolerance of termination criterion (default: 0.001)
+
+.TP
+.B -b, --backend arg
+choose the backend: automatic|openmp|cuda|hip|opencl|sycl (default: automatic)
```
[the remark this answers is not in the document]
Comment (breyerml) [reply docs/plssvm-train.1]: Closed via PR #26

File: docs/plssvm-train.1 [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+.TP
+.B -p, --target_platform arg
```
[the remark this answers is not in the document]
Comment (breyerml) [reply docs/plssvm-train.1]: Closed via PR #25

File: docs/plssvm-predict.1 [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+.SH OPTIONS
+
+.TP
+.B -b, --backend arg
+choose the backend: automatic|openmp|cuda|hip|opencl|sycl (default: automatic)
+
+.TP
+.B -p, --target_platform arg
+choose the target platform: automatic|cpu|gpu_nvidia|gpu_amd|gpu_intel (default: automatic)
```
[the remark this answers is not in the document]
Comment (breyerml) [reply docs/plssvm-predict.1]: Closed via PR #25

**Pull request merged**